src/Formats/MySQLBlockInputStream.cpp

```diff
@@ -30,6 +30,29 @@ namespace ErrorCodes
     extern const int NOT_IMPLEMENTED;
 }
 
+StreamSettings::StreamSettings(const Settings & settings)
+{
+    max_read_mysql_rows = settings.max_read_mysql_rows;
```

Comment:
By default it should be `settings.max_block_size` (because it is already used). Therefore, it is better to make `max_read_mysql_rows` disabled by default and used only if it was explicitly added.

Reply:
@TCeason

Reply:
> By default it should be settings.max_block_size (because it is already used). Therefore, it is better to make max_read_mysql_rows disabled by default and used only if it was explicitly added.

No, it does not be used. Before this pr the config max_block_size always equals 65505.

If the settings max_block_size has been modified to another value, the config `max_block_size` in 

MySQLBlockInputStream will not be modified. So we can modify its names to clear this means.

Reply:
No, it is passed to `MySQLBlockInputStream` from [here](https://github.com/ClickHouse/ClickHouse/blob/master/src/Interpreters/InterpreterSelectQuery.cpp#L1723) , after initialized [here](https://github.com/ClickHouse/ClickHouse/blob/master/src/Interpreters/InterpreterSelectQuery.cpp#L1602).

Reply:
I know what your means. But in MySQL storage engine read I will only use the new settings.

https://github.com/ClickHouse/ClickHouse/pull/22697/files#diff-fa68d1f80c3710c84f914aca3f7e2ea42c9e0a3571fe3baec5a7a3b44a90cd1aR74

I think it's suitable to control mysql data use only mysql settings.

Reply:
>  But in MySQL storage engine read I will only use the new settings.

Yes, I saw that. This is what I mean.

> I think it's suitable to control mysql data use only mysql settings.

But `max_block_size` is already used by people and they might expect that it is also used for mysql storage/dictionary as it always has been. And it does not always make sense to add one more setting specific to mysql. What I suggest is to use `max_read_mysql_rows` only if its value was explicitly changed, otherwise `max_block_size` should be used. 

Also it is much better to rename your settings to something like `external_storage_max_read_rows` and `external_storage_max_read_bytes`. So that they can also be used for other engines.

Reply:
> Also it is much better to rename your settings to something like external_storage_max_read_rows and external_storage_max_read_bytes. So that they can also be used for other engines.

Yes. your suggestion is more common that can cover mostly solution.

I think now I understand your means. If some users config `external_storage_max_read_rows` then the MySQL Storage will use this config. But if `external_storage_max_read_rows` is disabled the user will use `max_block_size`.

But I can not make `external_storage_max_read_rows` disable by default. Because it is also the config that MaterializeMySQL used. So in my mind the default value `settings.external_storage_max_read_rows` is equal with the default value `settings.max_block_size`.

Reply:
https://github.com/ClickHouse/ClickHouse/pull/22697/commits/e2c56f2705536aa7500e6a418ebb4ac00c004d0b#diff-fa68d1f80c3710c84f914aca3f7e2ea42c9e0a3571fe3baec5a7a3b44a90cd1aR98

Please reivew.